### PR TITLE
update kafka notebook with api changes and online training methodologies

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR updates the kafka tutorial notebook on https://www.tensorflow.org/io/tutorials/kafka by:
- updating the required API's
- adding a section for online training of models using `streaming.KafkaBatchIODataset`.

### CODE REVIEW

**Comment by kvignesh1420** [comment]
cc: @MarkDaoust @yongtang

**Comment by yongtang** [comment]
The PR LGTM. I also walked through on Colab and it works fine.

File: docs/tutorials/kafka.ipynb [hunk 1]
```diff
@@ -609,9 +609,9 @@
         "id": "ZPy0Ka21QII5"
       },
       "source": [
-        "**NOTE:** Please do not confuse the training step with online training. It's an entirely different paradigm.\n",
+        "**NOTE:** Please do not confuse the training step with online training. It's an entirely different paradigm which will be covered in a later section.\n",
```
Comment (lamberta) [inline docs/tutorials/kafka.ipynb:612]:
```suggestion
        "Note: Please do not confuse the training step with online training. It's an entirely different paradigm which will be covered in a later section.\n",
```

'Note:' is a style for the callout box on the site

File: docs/tutorials/kafka.ipynb [hunk 2]
```diff
@@ -717,6 +735,135 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IGph8eP9isuW"
+      },
+      "source": [
+        "The incrementally trained model can be saved in a periodic fashion (based on use-cases) and can be utilized to infer on the test data in either online or offline modes.\n",
+        "\n",
+        "**NOTE:** The `streaming.KafkaBatchIODataset` and `streaming.KafkaGroupIODataset` are still in experimental phase and have scope for improvements based on user-feedback."
```
Comment (lamberta) [inline docs/tutorials/kafka.ipynb:864]:
```suggestion
        "Note: The `streaming.KafkaBatchIODataset` and `streaming.KafkaGroupIODataset` are still in experimental phase and have scope for improvements based on user-feedback."
```

**Review by lamberta — APPROVED** [review]
A few formatting nits.
Confirmed it runs in Colab. Thanks

**Comment by kvignesh1420** [comment]
Made the changes @lamberta @yongtang.

**Comment by kvignesh1420** [comment]
closes https://github.com/tensorflow/io/issues/1076

**Pull request merged**